src/lib/generateDefinition.ts

```
@@ -4,92 +4,178 @@ import {getClusterAttributeValue} from './utils';
     return {externalDefinitionSource, definition};
 }
 
+function stringifyEps(endpoints: Endpoint[]): string[] {
+    return endpoints.map((e) => e.ID.toString());
+}
+
 const inputExtenders: Extender[] = [
-    [['msTemperatureMeasurement'], async (endpoint) => [{extend: m.temperature(), source: 'temperature()'}]],
-    [['msPressureMeasurement'], async (endpoint) => [{extend: m.pressure(), source: 'pressure()'}]],
-    [['msRelativeHumidity'], async (endpoint) => [{extend: m.humidity(), source: 'humidity()'}]],
-    [['msCO2'], async (endpoint) => [{extend: m.co2(), source: 'co2()'}]],
-    [['genPowerCfg'], async (endpoint) => [{extend: m.batteryPercentage(), source: 'batteryPercentage()'}]],
+    [['msTemperatureMeasurement'], async (eps) => [{extendFn: m.temperature, args: {endpoints: stringifyEps(eps)}, source: 'temperature'}]],
+    [['msPressureMeasurement'], async (eps) => [{extendFn: m.pressure, args: {endpoints: stringifyEps(eps)}, source: 'pressure'}]],
+    [['msRelativeHumidity'], async (eps) => [{extendFn: m.humidity, args: {endpoints: stringifyEps(eps)}, source: 'humidity'}]],
+    [['msCO2'], async (eps) => [{extendFn: m.co2, args: {endpoints: stringifyEps(eps)}, source: 'co2'}]],
+    [['genPowerCfg'], async () => [{extendFn: m.batteryPercentage, source: 'batteryPercentage'}]],
     [['genOnOff', 'lightingColorCtrl'], extenderOnOffLight],
     [['seMetering', 'haElectricalMeasurement'], extenderElectricityMeter],
     [['closuresDoorLock'], extenderLock],
 ];
 
 const outputExtenders: Extender[] = [
-    [['genIdentify'], async (endpoint) => [{extend: m.identify(), source: 'identify()'}]],
+    [['genIdentify'], async () => [{extend: m.identify(), source: 'identify()'}]],
 ];
 
-async function extenderLock(endpoint: Zh.Endpoint): Promise<GeneratedExtend[]> {
+async function extenderLock(endpoints: Zh.Endpoint[]): Promise<GeneratedExtend[]> {
+    // TODO: Support multiple endpoints
+    const endpoint = endpoints[0];
```

Please throw an error if multiple endpoints are passed

---

Updated, please check.